source/MRViewer/ImGuiHelpers.cpp

```diff
@@ -338,110 +338,142 @@ bool BeginStatePlugin( const char* label, bool* open, float width )
     return Begin( label, open, flags );
 }
 
-bool BeginCustomStatePlugin( const char* label, bool* open, bool* collapsed, float width, float scaling, float height, ImGuiWindowFlags flags )
+bool BeginCustomStatePlugin( const char* label, bool* open, bool* collapsed, float width, float scaling, float height, ImGuiWindowFlags flags, ImVec2* changedSize )
 {
+    const float closeButtonShift = 1.0f * scaling;
+    const auto& style = ImGui::GetStyle();    
+
+    const float titleBarHeight = 2 * MR::cRibbonItemInterval * scaling + ImGui::GetTextLineHeight() + closeButtonShift;
     if ( collapsed && *collapsed )
-        height = 0.0f;
+        height = titleBarHeight;
 
     ImGuiWindow* window = FindWindowByName( label );
-    auto menu = MR::getViewerInstance().getMenuPluginAs<MR::RibbonMenu>();
     if ( !window )
     {
+        auto menu = MR::getViewerInstance().getMenuPluginAs<MR::RibbonMenu>();
         float yPos = 0.0f;       
         if ( menu )
             yPos = menu->getTopPanelOpenedHeight() * menu->menu_scaling();
         SetNextWindowPos( ImVec2( GetIO().DisplaySize.x - width, yPos ), ImGuiCond_FirstUseEver );
     }
 
-    SetNextWindowSize( ImVec2( width, height ) );
-    const auto constriants = ( height == 0.0f ) ? ImVec2{ width, -1.0f } : ImVec2{ width, height };
-    SetNextWindowSizeConstraints( constriants, constriants );
+    if ( changedSize )
```

Comment:
mb try something like instead of new arg
```c++
ImVec2 changedSize;
if ( window && !bool(flags & No_Resize) )
    changedSize = ImVec2( window->Rect().GetWidth(), window->Rect().GetHeight() )

Reply:
It will not work because window forgets its full size after collapsing